[SPI Checking] Add partial SDKDBs for current iOS SDK and enable analysis

emw-apple opened the pull request:
#### 8a618fabde373e1ff595a572d74c83ce0f2ed4f9
<pre>
[SPI Checking] Add partial SDKDBs for current iOS SDK and enable analysis
<a href="https://bugs.webkit.org/show_bug.cgi?id=291613">https://bugs.webkit.org/show_bug.cgi?id=291613</a>
<a href="https://rdar.apple.com/149358945">rdar://149358945</a>

Reviewed by Sam Sneddon and Jonathan Bedard.

Add partial SDKDB records for iOS 18.2. These are generated using `tapi
sdkdb` from the public SDK image, and are named according to the public
libraries they represent (though, for brevity, the binary info is
stripped from SDKDB content).

This is a lot of data to check in, but as JSON blobs, SDKDBs compress
well (the objects are 2.3MB compared to 24MB in tree), and are
relatively stable between OS versions. For example, while this
introductory patch is 333K LOC, the patch to update for iOS 18.4 is
10824 insertions(+), 2430 deletions(-).

Enable auditing for open-source iOS builds and change some audit-spi
invocation logic to make it possible to audit with newer versions of iOS
than the one we&apos;ve checked in records for.

* Configurations/CommonBase.xcconfig:
* Configurations/WebKitProjectPaths.xcconfig:
* Source/WTF/Scripts/audit-spi-if-needed.sh:
* WebKitLibraries/SDKDBs/iphoneos18.2/: Lots of partial SDKDBs added.

Canonical link: <a href="https://commits.webkit.org/294570@main">https://commits.webkit.org/294570@main</a>
</pre>
<!--EWS-Status-Bubble-Start-->
https://github.com/WebKit/WebKit/commit/eb82148dfc5e0f396ca098fa044d72dd911a05ff

| Misc | iOS, visionOS, tvOS & watchOS  | macOS  | Linux |  Windows |
| ----- | ---------------------- | ------- |  ----- |  --------- |
| [✅ 🧪 style](https://ews-build.webkit.org/#/builders/38/builds/102234 "Passed style check") | [✅ 🛠 ios](https://ews-build.webkit.org/#/builders/131/builds/21902 "Built successfully") | [✅ 🛠 mac](https://ews-build.webkit.org/#/builders/138/builds/12217 "Built successfully") | [✅ 🛠 wpe](https://ews-build.webkit.org/#/builders/5/builds/107393 "Built successfully") | [✅ 🛠 win](https://ews-build.webkit.org/#/builders/59/builds/52870 "Built successfully") 
| [✅ 🧪 bindings](https://ews-build.webkit.org/#/builders/9/builds/104273 "Passed tests") | [✅ 🛠 ios-sim](https://ews-build.webkit.org/#/builders/130/builds/22210 "Built successfully") | [✅ 🛠 mac-AS-debug](https://ews-build.webkit.org/#/builders/123/builds/30409 "Built successfully") | [✅ 🧪 wpe-wk2](https://ews-build.webkit.org/#/builders/34/builds/77794 "Passed tests") | [✅ 🧪 win-tests](https://ews-build.webkit.org/#/builders/60/builds/34785 "Passed tests") 
| [✅ 🧪 webkitperl](https://ews-build.webkit.org/#/builders/11/builds/105240 "Passed tests") | [✅ 🧪 ios-wk2](https://ews-build.webkit.org/#/builders/132/builds/17188 "Passed tests") | [✅ 🧪 api-mac](https://ews-build.webkit.org/#/builders/18/builds/92291 "Passed tests") | [✅ 🧪 api-wpe](https://ews-build.webkit.org/#/builders/41/builds/58130 "Passed tests") | 
| | [✅ 🧪 ios-wk2-wpt](https://ews-build.webkit.org/#/builders/133/builds/17018 "Passed tests") | [✅ 🧪 mac-wk1](https://ews-build.webkit.org/#/builders/135/builds/10319 "Passed tests") | [✅ 🛠 wpe-cairo](https://ews-build.webkit.org/#/builders/65/builds/52228 "Built successfully") | 
| [✅ 🛠 🧪 jsc](https://ews-build.webkit.org/#/builders/20/builds/94905 "Built successfully and passed tests") | [✅ 🧪 api-ios](https://ews-build.webkit.org/#/builders/13/builds/86852 "Passed tests") | [✅ 🧪 mac-wk2](https://ews-build.webkit.org/#/builders/136/builds/10389 "Passed tests") | [✅ 🛠 gtk](https://ews-build.webkit.org/#/builders/2/builds/109769 "Built successfully") | 
| [✅ 🛠 🧪 jsc-arm64](https://ews-build.webkit.org/#/builders/12/builds/100843 "Built successfully and passed tests") | [✅ 🛠 vision](https://ews-build.webkit.org/#/builders/128/builds/29366 "Built successfully") | [✅ 🧪 mac-AS-debug-wk2](https://ews-build.webkit.org/#/builders/122/builds/21649 "Passed tests") | [✅ 🧪 gtk-wk2](https://ews-build.webkit.org/#/builders/1/builds/86775 "Passed tests") | 
| | [✅ 🛠 vision-sim](https://ews-build.webkit.org/#/builders/121/builds/29729 "Built successfully") | [✅ 🧪 mac-wk2-stress](https://ews-build.webkit.org/#/builders/8/builds/88494 "Passed tests") | [✅ 🧪 api-gtk](https://ews-build.webkit.org/#/builders/21/builds/86359 "Passed tests") | 
| [✅ 🛠 🧪 merge](https://ews-build.webkit.org/#/builders/19/builds/21986 "Built successfully and passed tests") | [✅ 🧪 vision-wk2](https://ews-build.webkit.org/#/builders/126/builds/31169 "Passed tests") | [❌ 🧪 mac-intel-wk2](https://ews-build.webkit.org/#/builders/137/builds/8887 "Found 2 new test failures: pageoverlay/overlay-small-frame-mouse-events.html pageoverlay/overlay-small-frame-paints.html (failure)") | [✅ 🛠 playstation](https://ews-build.webkit.org/#/builders/134/builds/23608 "Built successfully") | 
| | [✅ 🛠 tv](https://ews-build.webkit.org/#/builders/127/builds/29294 "Built successfully") | [✅ 🛠 mac-safer-cpp](https://ews-build.webkit.org/#/builders/120/builds/34589 "Built successfully") | [✅ 🛠 jsc-armv7](https://ews-build.webkit.org/#/builders/35/builds/124469 "Built successfully") | 
| | [✅ 🛠 tv-sim](https://ews-build.webkit.org/#/builders/125/builds/29105 "Built successfully") | | [✅ 🧪 jsc-armv7-tests](https://ews-build.webkit.org/#/builders/25/builds/34561 "Passed tests") | 
| | [✅ 🛠 watch](https://ews-build.webkit.org/#/builders/129/builds/32428 "Built successfully") | | | 
| | [✅ 🛠 watch-sim](https://ews-build.webkit.org/#/builders/124/builds/30664 "Built successfully") | | | 
<!--EWS-Status-Bubble-End-->

gsnedders reviewed (changes requested):
> Here's what that patch will actually look like: [emw-apple/WebKit@emw-apple:eng/SPI-Checking-Add-partial-SDKDBs-for-current-iOS-SDK-and-enable-analysis...emw-apple:WebKit:eng/spi-checking-iphoneos18.4-sdkdb](https://github.com/emw-apple/WebKit/compare/emw-apple:eng/SPI-Checking-Add-partial-SDKDBs-for-current-iOS-SDK-and-enable-analysis...emw-apple:WebKit:eng/spi-checking-iphoneos18.4-sdkdb)

I don't think the diff is awful.

To note, of the -2.5k deletions:

368 are `-   "target": "arm64{,e}-apple-ios18.2"`.

Some of the changes, like `AVPlayerMediaSelectionCriteria`'s `initWithPreferredLanguages:preferredMediaCharacteristics:` seem like they're purely reordering?

The biggest category of changes, and the one that slightly worries me, seems to be globals removing `"unavailable": true`. We don't seem to ever read `unavailable`, and I'm not totally sure what it refers to — is this just symbols marked as unavailable on that platform? If so, we can likely filter them out while they're unavailable, because it effectively means they don't exist. Otherwise, we should probably be checking they're unused?

We should figure out the answer to that last point before landing this, IMO — as otherwise we might be bogusly allowing SPI?

emw-apple committed:
[SPI Checking] Add partial SDKDBs for current iOS SDK and enable analysis
https://bugs.webkit.org/show_bug.cgi?id=291613
rdar://149358945

Reviewed by Sam Sneddon and Jonathan Bedard.

Add partial SDKDB records for iOS 18.2. These are generated using `tapi
sdkdb` from the public SDK image, and are named according to the public
libraries they represent (though, for brevity, the binary info is
stripped from SDKDB content).

This is a lot of data to check in, but as JSON blobs, SDKDBs compress
well (the objects are 2.3MB compared to 24MB in tree), and are
relatively stable between OS versions. For example, while this
introductory patch is 333K LOC, the patch to update for iOS 18.4 is
10824 insertions(+), 2430 deletions(-).

Enable auditing for open-source iOS builds and change some audit-spi
invocation logic to make it possible to audit with newer versions of iOS
than the one we've checked in records for.

* Configurations/CommonBase.xcconfig:
* Configurations/WebKitProjectPaths.xcconfig:
* Source/WTF/Scripts/audit-spi-if-needed.sh:
* WebKitLibraries/SDKDBs/iphoneos18.2/: Lots of partial SDKDBs added.

Canonical link: https://commits.webkit.org/294570@main
